integration/query_compat_test.go

```diff
@@ -200,8 +208,9 @@ func TestQueryCompatFilter(t *testing.T) {
 			resultPushdown: true,
 		},
 		"IDObjectID": {
-			filter:         bson.D{{"_id", primitive.NilObjectID}},
-			resultPushdown: true,
+			filter:               bson.D{{"_id", primitive.NilObjectID}},
```

Comment:
If we want to enable this flag in tests and check the logic, do we have a test case where such pushdown is applied to a non-nil object id? Just some other object id that we store in the DB? E.g. `objectid` doc from `ObjectIDs` collection or any other.

Reply:
Let's add one. Looks like existing tests are all nil objectID